misk-core/src/main/kotlin/misk/logging/SampledLogger.kt

```diff
@@ -0,0 +1,1136 @@
+   */
+  override fun info(marker: org.slf4j.Marker?, msg: String?, t: Throwable?) {
+    if (sampler.sample()) {
+      logger.info(marker, msg, t)
+    }
+  }
+
+  /**
+   * Is the logger instance enabled for the DEBUG level?
+   *
+   * @return True if this Logger is enabled for the DEBUG level,
+   * false otherwise.
+   */
+  override fun isDebugEnabled(): Boolean {
+    return logger.isDebugEnabled
+  }
+
+  /**
+   * Similar to [.isDebugEnabled] method except that the
+   * marker data is also taken into account.
+   *
+   * @param marker The marker data to take into consideration
+   * @return True if this Logger is enabled for the DEBUG level,
+   * false otherwise.
+   */
+  override fun isDebugEnabled(marker: org.slf4j.Marker?): Boolean {
+    return logger.isDebugEnabled(marker)
+  }
+
+  /**
+   * Is the logger instance enabled for the ERROR level?
+   *
+   * @return True if this Logger is enabled for the ERROR level,
+   * false otherwise.
+   */
+  override fun isErrorEnabled(): Boolean {
+    return logger.isErrorEnabled
+  }
+
+  /**
+   * Similar to [.isErrorEnabled] method except that the
+   * marker data is also taken into consideration.
+   *
+   * @param marker The marker data to take into consideration
+   * @return True if this Logger is enabled for the ERROR level,
+   * false otherwise.
+   */
+  override fun isErrorEnabled(marker: org.slf4j.Marker?): Boolean {
+    return logger.isErrorEnabled(marker)
+  }
+
+  /**
+   * Is the logger instance enabled for the INFO level?
+   *
+   * @return True if this Logger is enabled for the INFO level,
+   * false otherwise.
+   */
+  override fun isInfoEnabled(): Boolean {
+    return logger.isInfoEnabled()
+  }
+
+  /**
+   * Similar to [.isInfoEnabled] method except that the marker
+   * data is also taken into consideration.
+   *
+   * @param marker The marker data to take into consideration
+   * @return true if this logger is warn enabled, false otherwise
+   */
+  override fun isInfoEnabled(marker: org.slf4j.Marker?): Boolean {
+    return logger.isInfoEnabled(marker)
+  }
+
+  /**
+   * Is the logger instance enabled for the TRACE level?
+   *
+   * @return True if this Logger is enabled for the TRACE level,
+   * false otherwise.
+   * @since 1.4
+   */
+  override fun isTraceEnabled(): Boolean {
+    return logger.isTraceEnabled()
+  }
+
+  /**
+   * Similar to [.isTraceEnabled] method except that the
+   * marker data is also taken into account.
+   *
+   * @param marker The marker data to take into consideration
+   * @return True if this Logger is enabled for the TRACE level,
+   * false otherwise.
+   *
+   * @since 1.4
+   */
+  override fun isTraceEnabled(marker: org.slf4j.Marker?): Boolean {
+    return logger.isTraceEnabled(marker)
+  }
+
+  /**
+   * Is the logger instance enabled for the WARN level?
+   *
+   * @return True if this Logger is enabled for the WARN level,
+   * false otherwise.
+   */
+  override fun isWarnEnabled(): Boolean {
```

Comment:
For methods like these (and there are quite a few) that don't actually get overridden with sampling, I think it's better for you to use delegation rather than implementing here https://kotlinlang.org/docs/reference/delegation.html.